### PR TITLE
docs: add doc for multitask fine-tuning

### PR DESCRIPTION
**Pull request by iProzd**
Add docs for multitask fine-tuning.

## Summary by CodeRabbit

- **Documentation**
	- Updated the fine-tuning guide with new sections on TensorFlow and PyTorch implementations.
	- Added detailed instructions for fine-tuning methods in PyTorch, including specific commands and configurations.
	- Modified the multi-task training guide to redirect users to the fine-tuning section for more comprehensive instructions.
	- Corrected a typo in the multi-task training TensorFlow documentation for improved clarity.

### CODE REVIEW

File: doc/train/finetuning.md [hunk 1]
```diff
@@ -45,3 +47,121 @@ To obtain a more simplified script, for example, you can change the {ref}`model
...
+
+```bash
+$ dp --pt train input.json --finetune pretrained.pt
+```
+
+The model section in input.json can be simplified as follows:
+
+```json
+    "model": {
+        "type_map":     ["O", "H"],
```
Comment (wanghan-iapcm) [inline doc/train/finetuning.md:70]: warn explicitly after the example that the type map is overwritten by the pretrained model, and say the behavior may be changed in beta release.

File: doc/train/finetuning.md [hunk 2]
```diff
@@ -45,3 +47,121 @@ To obtain a more simplified script, for example, you can change the {ref}`model
...
+        "descriptor" :  {},
+        "fitting_net" : {}
+    }
+```
+
+Additionally, within the PyTorch implementation, we also support more general multitask pre-trained models,
+which includes multiple datasets for pre-training. These pre-training datasets share a common descriptor while maintaining their individual fitting nets,
+as detailed in the DPA2 [paper](https://arxiv.org/abs/2312.15492).
+For fine-tuning using this multitask pre-trained model (`multitask_pretrained.pt`),
+one can select a specific branch (e.g., `CHOOSEN_BRANCH`) included in `multitask_pretrained.pt` for fine-tuning with the following command:
```
Comment (wanghan-iapcm) [inline doc/train/finetuning.md:80]: how to check the available branches?

File: doc/train/finetuning.md [hunk 3]
```diff
@@ -45,3 +47,121 @@ To obtain a more simplified script, for example, you can change the {ref}`model
...
+
+## PyTorch Implementation {{ pytorch_icon }}
+
+In PyTorch version, leveraging the flexibility offered by the framework and the multi-task training capabilities provided by DPA2 [paper](https://arxiv.org/abs/2312.15492),
+we have introduced an updated, more adaptable approach to fine-tuning. This methodology encompasses two primary variations:
+
+### 1. Single-task fine-tuning
+
+This fine-tuning method is similar to the fine-tune approach supported by TensorFlow.
+It utilizes a standard pre-trained model (`pretrained.pt`, single-task pre-trained) and modifies the energy bias within its fitting net before continuing with training.
```
Comment (wanghan-iapcm) [inline doc/train/finetuning.md:59]: what is a "standard" pre-trained model?

Comment (wanghan-iapcm) [reply doc/train/finetuning.md:59]: distinguish the two cases:
1. fine tune from single task pretrained model 
2. fine tune from multi-task pretrained model.

File: doc/train/finetuning.md [hunk 4]
```diff
@@ -45,3 +47,121 @@ To obtain a more simplified script, for example, you can change the {ref}`model
         "fitting_net" : {}
     }
 ```
+
+## PyTorch Implementation {{ pytorch_icon }}
+
+In PyTorch version, leveraging the flexibility offered by the framework and the multi-task training capabilities provided by DPA2 [paper](https://arxiv.org/abs/2312.15492),
```
Comment (wanghan-iapcm) [inline doc/train/finetuning.md:53]: by saying " multi-task training capabilities " and referring to DPA-2 paper,   the reader may think you assume that the model is pretrained by multi-task in this section

File: doc/train/finetuning.md [hunk 5]
```diff
@@ -45,3 +47,121 @@ To obtain a more simplified script, for example, you can change the {ref}`model
...
+we have introduced an updated, more adaptable approach to fine-tuning. This methodology encompasses two primary variations:
+
+### 1. Single-task fine-tuning
+
+This fine-tuning method is similar to the fine-tune approach supported by TensorFlow.
+It utilizes a standard pre-trained model (`pretrained.pt`, single-task pre-trained) and modifies the energy bias within its fitting net before continuing with training.
+The command for this operation is:
+
+```bash
+$ dp --pt train input.json --finetune pretrained.pt
```
Comment (wanghan-iapcm) [inline doc/train/finetuning.md:63]: how to finetune from a randomly initialized fitting net?

**Review by wanghan-iapcm — CHANGES_REQUESTED** [review]

File: doc/train/finetuning.md [hunk 6]
```diff
@@ -45,3 +47,121 @@ To obtain a more simplified script, for example, you can change the {ref}`model
         "fitting_net" : {}
     }
 ```
+
+## PyTorch Implementation {{ pytorch_icon }}
+
+In PyTorch version, leveraging the flexibility offered by the framework and the multi-task training capabilities provided by DPA2 [paper](https://arxiv.org/abs/2312.15492),
+we have introduced an updated, more adaptable approach to fine-tuning. This methodology encompasses two primary variations:
+
+### 1. Single-task fine-tuning
```
Comment (njzjz) [inline doc/train/finetuning.md:56]: `1.` is not required. Sphinx will generate it.

File: doc/train/finetuning.md [hunk 7]
```diff
@@ -45,3 +47,121 @@ To obtain a more simplified script, for example, you can change the {ref}`model
...
+as detailed in the DPA2 [paper](https://arxiv.org/abs/2312.15492).
+For fine-tuning using this multitask pre-trained model (`multitask_pretrained.pt`),
+one can select a specific branch (e.g., `CHOOSEN_BRANCH`) included in `multitask_pretrained.pt` for fine-tuning with the following command:
+
+```bash
+$ dp --pt train input.json --finetune multitask_pretrained.pt --model-branch CHOOSEN_BRANCH
+```
+
+This command will start fine-tuning based on the pre-trained model's descriptor and the selected branch's fitting net.
+If --model-branch is not set, a randomly initialized fitting net will be used.
```
Comment (njzjz) [inline doc/train/finetuning.md:87]: It is a duplicate of Section 5.6.3. We may only maintain one and provide a link from another one.

File: doc/train/finetuning.md [hunk 8]
```diff
@@ -45,3 +47,121 @@ To obtain a more simplified script, for example, you can change the {ref}`model
...
+which could be more beneficial for fine-tuning.
+
+To achieve this, it is first necessary to clearly identify the datasets from which the pre-trained model originates and to download the corresponding datasets
+that need to be retained for subsequent multitask fine-tuning.
+Then, prepare a suitable input script for multitask fine-tuning `multi_input.json` as the following steps.
+
+- Suppose the new dataset for fine-tuning is named `DOWNSTREAM_DATA`, and the datasets to be retained from multitask pre-trained model are `PRE_DATA1` and `PRE_DATA2`. One can:
+
+1. Refer to the [`multi-task-training`](./multi-task-training-pt.md) document to prepare a multitask training script for two systems,
+   ideally extracting parts (i.e. `model_dict`, `loss_dict`, `data_dict` and `model_prob` parts) corresponding to `PRE_DATA1` and `PRE_DATA2` directly from the training script of the pre-trained model.
```
Comment (njzjz) [inline doc/train/finetuning.md:102]: We may link `loss_dict`, etc, to the corresponding parameter section.

File: doc/train/finetuning.md (hunk 5 above)
Comment (iProzd) [reply doc/train/finetuning.md:66]: We do not yet support fine-tuning from a randomly initialized fitting net in single-task fine-tuning from single-task pretrained model, which is the same as implementations in TensorFlow. I've opened an [issue](https://github.com/deepmodeling/deepmd-kit/issues/3741) for this.

File: doc/train/finetuning.md (hunk 2 above)
Comment (iProzd) [reply doc/train/finetuning.md:95]: We do not have explicit command for this yet, I've opened an [issue](https://github.com/deepmodeling/deepmd-kit/issues/3742) for this.

**Review by njzjz — APPROVED** [review]

**Review by wanghan-iapcm — APPROVED** [review]

**Pull request merged**